File: legacy/DPB_Comp/matlab/UserEquipment.m
classdef UserEquipment < Transceiver
    
    properties
        servingBaseStation = []
    end
    
    methods
        function newObject = UserEquipment(x, y)
            newObject = newObject@Transceiver(x, y);
        end
        
        function associateToBaseStation(thisObject, baseStation)
            thisObject.servingBaseStation = baseStation;
        end
        
        function dissociateFromBaseStations(thisObject)
            thisObject.servingBaseStation = [];
        end
        
        function baseStation = getServingBaseStation(thisObject)
            baseStation = thisObject.servingBaseStation;
        end
        
        function isDummy = isDummy(thisObject)
            [x, y] = thisObject.getPosition;
            isDummy = (x < 0) & (y < 0);
        end
    end
   
end

File: temp/UserEquipment.py
from Transceiver import Transceiver


class UserEquipment(Transceiver):
    def __init__(self, x, y):
        super().__init__(x, y)
        self.serving_base_station = None

    def associate_to_base_station(self, base_station):
        self.serving_base_station = base_station

    def dissociate_from_base_stations(self):
        self.serving_base_station = None

    def get_serving_base_station(self):
        return self.serving_base_station

    def is_dummy(self):
        x, y = self.get_position()
        return x < 0 and y < 0
